### ISSUE TITLE
Unsupported mime type application/octet-stream when importing PDF to other application

### ISSUE THREAD

**Issue by psaletan** [dov/paps#57]
The PDF documents produced by the latest paps seem to lack some metadata, including MIME type. This isn't a problem for opening and reading them on my host (Ubuntu 20.04). However, when I feed them to paperless-ngx, it refuses to ingest them, complaining about unsupported mime type.

```
paul@pauldev:~/Downloads$ paps -v
Version: paps-0.7.9
```

Generate test document:
```
paps --header --header-left="{filename}" --header-center="" --header-right="Page {page_idx:01d}/{num_pages:01d}" --footer --footer-left="Modified {mtime:%Y-%m-%d %H:%M}"--footer-right="Printed {now:%Y-%m-%d %H:%M}" --paper=letter --separation-lines --bottom-margin=42 --font="Roboto 10" --format=pdf 'test.txt' > test.pdf
```

Paperless server log when importing the PDF:
```
[2023-02-10 13:47:52,082] [INFO] [paperless.consumer] Consuming test.pdf
[2023-02-10 13:47:52,151] [ERROR] [paperless.consumer] Unsupported mime type application/octet-stream
[2023-02-10 13:47:52,180] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[1f49b3f0-ea90-4fd5-bfbb-072e993ad954] raised unexpected: ConsumerError('test.pdf: Unsupported mime type application/octet-stream')
```

Exif tool shows metadata missing mime type:
```
paul@pauldev:~/Downloads$ exiftool test.pdf
ExifTool Version Number         : 11.88
File Name                       : test.pdf
Directory                       : .
File Size                       : 9.9 kB
File Modification Date/Time     : 2023:02:10 08:46:44-05:00
File Access Date/Time           : 2023:02:10 08:47:45-05:00
File Inode Change Date/Time     : 2023:02:10 08:46:44-05:00
File Permissions                : rw-rw-r--
Error                           : File format error
```

My workaround: Correct the PDF by piping/rewriting paps PDF through ghostscript:
```
paps --header --header-left="{filename}" --header-center="" --header-right="Page {page_idx:01d}/{num_pages:01d}" --footer --footer-left="Modified {mtime:%Y-%m-%d %H:%M}"--footer-right="Printed {now:%Y-%m-%d %H:%M}" --paper=letter --separation-lines --bottom-margin=42 --font="Roboto 10" --format=pdf test.txt - | gs -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE="test.pdf" -dBATCH -
GPL Ghostscript 9.50 (2019-10-15)
Copyright (C) 2019 Artifex Software, Inc.  All rights reserved.
This software is supplied under the GNU AGPLv3 and comes with NO WARRANTY:
see the file COPYING for details.
   **** Warning: File has some garbage before %PDF- .
Processing pages 1 through 1.
Page 1
```

Exif tool after rewriting:
```
paul@pauldev:~/Downloads$ exiftool test.pdf
ExifTool Version Number         : 11.88
File Name                       : test.pdf
Directory                       : .
File Size                       : 11 kB
File Modification Date/Time     : 2023:02:10 08:54:21-05:00
File Access Date/Time           : 2023:02:10 08:54:24-05:00
File Inode Change Date/Time     : 2023:02:10 08:54:21-05:00
File Permissions                : rw-rw-r--
File Type                       : PDF
File Type Extension             : pdf
MIME Type                       : application/pdf
PDF Version                     : 1.7
Linearized                      : No
Page Count                      : 1
XMP Toolkit                     : XMP toolkit 2.9.1-13, framework 1.6
About                           : uuid:f7758c34-e166-11f8-0000-0bc720dc2f76
Producer                        : GPL Ghostscript 9.50
Modify Date                     : 2023:02:10 08:54:20-05:00
Create Date                     : 2023:02:10 08:54:20-05:00
Creator Tool                    : UnknownApplication
Document ID                     : uuid:f7758c34-e166-11f8-0000-0bc720dc2f76
Format                          : application/pdf
Title                           : Untitled
```

Paps is a great tool, and worth the effort to compile. I struggled to convert non-markdown text files to PDF using Pandoc. Paps is several times faster and supplies simple parameters to set the options I want. I hope you can add metadata entries to satisfy apps like paperless.

**Comment by dov** [comment]
Thanks for the feedback. Indeed I wrote some garbage to stdout in version 0.7.9. Yesterday I released a new version 0.8.0, with the latest version exiftool outputs:
```
ExifTool Version Number         : 12.42
File Name                       : test.pdf
Directory                       : .
File Size                       : 9.8 kB
File Modification Date/Time     : 2023:02:10 16:14:05+02:00
File Access Date/Time           : 2023:02:10 16:13:46+02:00
File Inode Change Date/Time     : 2023:02:10 16:14:05+02:00
File Permissions                : -rw-r--r--
File Type                       : PDF
File Type Extension             : pdf
MIME Type                       : application/pdf
PDF Version                     : 1.7
Linearized                      : No
Page Count                      : 1
Producer                        : cairo 1.17.6 (https://cairographics.org)
Create Date                     : 2023:02:10 16:14:05+02:00
```
Not as rich as the ghostscript output, but perhaps it is enough.

**Comment by psaletan** [comment]
Thanks. I should have looked for the new release before posting. I just installed 0.8.0 and the problem is fixed.
